app/src/androidTest/java/com/mapbox/mapboxsdk/plugins/locationlayer/LocationLayerPluginTest.kt

```diff
@@ -106,6 +106,29 @@ class LocationLayerPluginTest {
     executePluginTest(pluginAction, PluginGenerationUtil.getLocationLayerPluginProvider(rule.activity, true))
   }
 
+  @Test
+  fun locationLayerPlugin_initializesLocationEngineCorrectlyWhenOnesNotProvidedButHasOptions() {
+    val pluginAction = object : GenericPluginAction.OnPerformGenericPluginAction<LocationLayerPlugin> {
+      override fun onGenericPluginAction(plugin: LocationLayerPlugin, mapboxMap: MapboxMap,
+                                         uiController: UiController, context: Context) {
+
+        val locationEngine = plugin.locationEngine
+        assertThat(locationEngine, notNullValue())
+
+        uiController.loopMainThreadForAtLeast(MAP_CONNECTION_DELAY)
+        assertThat(locationEngine?.isConnected, `is`(true))
```

Comment:
It would be great to verify in this test if the options were initialized correctly as well.